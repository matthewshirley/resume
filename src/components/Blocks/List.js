import React from 'react';
import PropTypes from 'prop-types';

import { Box, Text } from '..';

const List = ({
  highlights,
}) => {
  const listOfHighlights = highlights.map((highlight, index) => (
    <Text key={index} as="li" m={0} my={2} fontSize={1}>{highlight}</Text>
  ));

  return (
    <Box as="ul" mt={2}>
      {listOfHighlights}
    </Box>
  );
};

List.propTypes = {
  highlights: PropTypes.arrayOf(
    PropTypes.string,
  ).isRequired,
};

export default List;
