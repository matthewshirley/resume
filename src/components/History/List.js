import React from 'react';
import PropTypes from 'prop-types';

import Box from '../common/Box';
import Text from '../common/Text';

const List = ({
  highlights,
}) => {
  const listOfHighlights = highlights.map((highlight, index) => (
    <Text key={index} as="li" m={0} my={2}>{highlight}</Text>
  ));

  return (
    <Box as="ul" mt={2} pl={{ sm: '1rem', lg: '2rem', print: '2rem'}}>
      {listOfHighlights}
    </Box>
  );
};

List.propTypes = {
  highlights: PropTypes.arrayOf(
    PropTypes.string,
  ).isRequired,
};

export default List;
