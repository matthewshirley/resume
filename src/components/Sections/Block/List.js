import React from 'react';
import PropTypes from 'prop-types';

import { Text } from '@mattshirley/design/src/components/Text';
import { Box } from '@mattshirley/design/src/components/Layout/Box';

const List = ({
  highlights,
}) => {
  const listOfHighlights = highlights.map((highlight, index) => (
    <Text key={index} as="li" fontSize={1} my={2}>{highlight}</Text>
  ));

  return (
    <Box as="ul" mt={2}>
      {listOfHighlights}
    </Box>
  );
};

List.propTypes = {
  highlights: [
    PropTypes.string.isRequired,
  ].isRequired,
};

export default List;
