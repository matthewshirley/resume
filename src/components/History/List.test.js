import React from 'react';
import renderer from 'react-test-renderer';

import List from './List';


describe('List', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<List highlights={['a', 'b', 'c']} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
