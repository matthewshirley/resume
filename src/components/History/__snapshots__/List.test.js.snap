// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly 1`] = `
<ul
  className="sc-AxjAm jKDKBO"
>
  <li
    className="sc-AxirZ ilUQMD"
    fontSize={1}
  >
    a
  </li>
  <li
    className="sc-AxirZ ilUQMD"
    fontSize={1}
  >
    b
  </li>
  <li
    className="sc-AxirZ ilUQMD"
    fontSize={1}
  >
    c
  </li>
</ul>
`;
